### PR TITLE
enhance: Remove pre-marking segments as L2 during clustering compaction

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #36686

master pr: #36799 

The core of this change is to **ensure that the many-to-many lineage derivation logic is correct, making sure that both the parent and child cannot simultaneously exist in the target segment view.**

feature:
  - Clustering compaction no longer marks the input segments as L2.
- Add a new field `is_invisible` to `segmentInfo`, and mark segments that have completed clustering but have not yet built indexes as `is_invisible` to prevent them from being loaded prematurely."
- Do not mark the input segment as `Dropped` before the clustering compaction is completed.
- After compaction fails, only the result segment needs to be marked as Dropped.

compatibility:
- If the upgraded task has not failed, there are no compatibility issues.
- If the status after the upgrade is `MetaSaved`, then skip the stats task based on whether TmpSegments is empty.
  - If the failure occurs before `MetaSaved`:
- there are no ResultSegments, and InputSegments have not been marked as dropped yet.
    - the level of input segments need to revert to LastLevel
  - If the failure occurs after `MetaSaved`:
- ResultSegments have already been generated, and InputSegments have been marked as Dropped. At this point, simply make the ResultSegments visible.
- the level of ResultSegments needs to be set to L1（in order to participate in mixCompaction）

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/datacoord/compaction_task_clustering.go [hunk 1]
```diff
@@ -307,28 +297,41 @@ func (t *clusteringCompactionTask) processIndexing() error {
...
-	for _, segID := range t.GetResultSegments() {
-		operators = append(operators, UpdateSegmentPartitionStatsVersionOperator(segID, t.GetPlanID()))
-	}
-	err = t.meta.UpdateSegmentsInfo(operators...)
-	if err != nil {
-		return merr.WrapErrClusteringCompactionMetaError("UpdateSegmentPartitionStatsVersion", err)
+	// mark input segments as dropped
+	// now, the segment view only includes the result segments.
+	if err = t.markInputSegmentsDropped(); err != nil {
+		return err
```
Comment (czs007) [inline internal/datacoord/compaction_task_clustering.go:334]: do not return error for now.

File: internal/datacoord/meta.go [hunk 2]
```diff
@@ -1408,14 +1412,9 @@ func (m *meta) completeClusterCompactionMutation(t *datapb.CompactionTask, resul
 		}
 
 		cloned := segment.Clone()
```
Comment (czs007) [inline internal/datacoord/meta.go:1414]: The cloned  here seems unnecessary.

Comment (czs007) [reply internal/datacoord/meta.go:1414]: Both compactFromSegIDs and compactFromSegInfos are unnecessary.

**Pull request merged**